From ac34f7cbd76ad8223261e9002e2fc8b9eff1879a Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Wed, 30 Dec 2020 18:40:19 +0100
Subject: [PATCH 06/21] Replace Make foreach function with addprefix when
 possible

The former does not temporarily define new variables, its strictly
functional effects are straightforward.
---
 debian/rules | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/debian/rules b/debian/rules
index 7d3b285..7b165c8 100755
--- a/debian/rules
+++ b/debian/rules
@@ -1010,7 +1010,7 @@ ifeq ($(with_hppa64),yes)
        build_stamps += stamps/build-hppa64
 endif
 ifeq ($(with_cross),yes)
-       build_stamps += $(foreach ca,$(CROSS_ARCHS), stamps/build.$(ca))
+       build_stamps += $(addprefix stamps/build.,$(CROSS_ARCHS))
 endif
 ifneq (,$(TARGET))
    ifeq ($(TARGET),hppa64-linux-gnu)
@@ -1041,7 +1041,7 @@ ifeq ($(with_hppa64),yes)
        install_stamps += stamps/install-hppa64
 endif
 ifeq ($(with_cross),yes)
-       install_stamps += $(foreach ca,$(CROSS_ARCHS), stamps/install.$(ca))
+       install_stamps += $(addprefix stamps/install.,$(CROSS_ARCHS))
 endif
 
 ifneq (,$(TARGET))
@@ -1538,7 +1538,7 @@ binary-for-host: checkroot
 	$(DPKG_DEB) --build $(d_hst) ..
 
 binary-arch: checkroot binary-native binary-for-host \
-    $(if $(filter yes, $(with_cross)),$(foreach ca,$(CROSS_ARCHS), binary.$(ca)))
+    $(if $(filter yes, $(with_cross)),$(addprefix binary.,$(CROSS_ARCHS)))
 	$(checkdir)
 
 binary-native: checkroot build install $(checkdir)
-- 
2.30.1

